### PR TITLE
docs: add usage guide for the project

### PR DESCRIPTION
**Pull request by xixas**
Create usage-guide which covers using API for sending notifications

### CODE REVIEW

File: docs/usage-guide.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Usage Guide for Osmo-Notify
```
Comment (osm-vishnukyatannawar) [inline docs/usage-guide.md:1]: Add this in readme just like other 2 documents.

File: docs/usage-guide.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+Welcome to the usage guide for Osmo-Notify, a powerful notification management system designed to simplify the process of sending notifications through various channels. This guide will walk you through the steps of integrating Osmo-Notify into your application and making the most of its features.
+
+## Table of Contents
+
+1. Overview
+2. Pushing Data to the Database
+3. Using the Osmo-Notify API
+4. Tracking Notification Status
+5. Available Channel Types
+6. Delivery Status Information
```
Comment (osm-vishnukyatannawar) [inline docs/usage-guide.md:14]: Make this links to the section

File: docs/usage-guide.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+## 1. Overview
+
+Osmo-Notify offers a streamlined solution for sending notifications via different channels. It operates by monitoring the `notify_notifications` database table and automatically dispatches notifications for records with a `PENDING` status.
+
+## 2. Pushing Data to the Database
+
+Developers have the flexibility to populate the `notify_notifications` database table directly or opt for the recommended approach of utilizing the Osmo-Notify API. By following this method, the notification processing is seamlessly managed by Osmo-Notify.
```
Comment (osm-vishnukyatannawar) [inline docs/usage-guide.md:22]: You should give an example with a query.

And later we should have sections like for SMTP how to send it.

Have a section where we can list them and each channel will have its own page.

Where we have to explain what are the each and every property and what should be stored.

For this database insertation you have to show may be by adding a sql query.

With an example.

Like if someone want to just send hello world email to a user with some subject as hello world then how to do it.

Same example can be then used for API too.

But like I mentioned we need then a dedicated page for each channel.

File: docs/usage-guide.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+To use the Osmo-Notify API, follow these steps:
+
+- **Method:** POST
+- **Endpoint:** `/notifications`
+
+**Sample Request Body:**
+```json
+{
+  "channelType": 1,
```
Comment (osm-vishnukyatannawar) [inline docs/usage-guide.md:34]: is channel type a enum now? I think we are sending text here.

File: docs/usage-guide.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    "modifiedBy": null,
+    "modifiedOn": null,
+    "id": 1
+  }
+}
+```
+
+## 4. Tracking Notification Status
+
+Osmo-Notify updates the `deliveryStatus` and `result` columns to provide information on the notification's status. Use these columns to track the progress of your notifications.
```
Comment (osm-vishnukyatannawar) [inline docs/usage-guide.md:70]: if you refer to column then better use the exact column name is it not `delivery_status`?

File: docs/usage-guide.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+## 5. Available Channel Types
+
+Osmo-Notify supports multiple channel types, allowing you to choose the most suitable one for your notifications. Currently, the available channel types are:
+
+- **SMTP (Simple Mail Transfer Protocol):** ChannelType = 1
```
Comment (osm-vishnukyatannawar) [inline docs/usage-guide.md:76]: Add table and like I told above we need these as links in table which will take to that respective page.

There we should tell what are the env values associated with these and then we should tell the notification data structure. and explain each and every property.

File: docs/usage-guide.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+## 6. Delivery Status Information
+
+Osmo-Notify provides different delivery status options to reflect the state of your notifications:
+
+- **PENDING:** The notification is awaiting processing. DeliveryStatus = 1
```
Comment (osm-vishnukyatannawar) [inline docs/usage-guide.md:82]: Would like it to be a table

**Review by osm-vishnukyatannawar — CHANGES_REQUESTED** [review]

File: docs/usage-guide.md (hunk 1 above)
Comment (xixas) [reply docs/usage-guide.md:1]: Added

File: docs/usage-guide.md (hunk 2 above)
Comment (xixas) [reply docs/usage-guide.md]: Updated

File: docs/usage-guide.md (hunk 4 above)
Comment (xixas) [reply docs/usage-guide.md:34]: Channel type is enum(using tinyint)

File: docs/usage-guide.md (hunk 5 above)
Comment (xixas) [reply docs/usage-guide.md:70]: Update column name

File: docs/usage-guide.md (hunk 6 above)
Comment (xixas) [reply docs/usage-guide.md:76]: Added a table with link to document.

File: docs/usage-guide.md (hunk 7 above)
Comment (xixas) [reply docs/usage-guide.md:82]: Added table

**Review by osm-vishnukyatannawar — APPROVED** [review]

**Pull request merged**